Sources/Controllers/Panels/OAMapPanelViewController.mm

```diff
@@ -2778,18 +2778,15 @@ - (void)openTargetViewWithRouteDetailsGraph:(OAGPXDocument *)gpx
     _activeTargetType = targetPoint.type;
     _activeTargetObj = targetPoint.targetObj;
     _targetMenuView.activeTargetType = _activeTargetType;
+    _targetMenuView.trackMenuDelegate = trackMenuDelegate;
 
     [_targetMenuView setTargetPoint:targetPoint];
     [self applyTargetPoint:targetPoint];
 
     [self enterContextMenuMode];
     [self showTargetPointMenu:NO showFullMenu:NO onComplete:^{
-        if (trackMenuDelegate)
-        {
-            OARouteDetailsGraphViewController *graphController = (OARouteDetailsGraphViewController *) self.targetMenuView.customController;
-            graphController.trackMenuDelegate = trackMenuDelegate;
-            [graphController onNewModeSelected:modeType];
-        }
+        if ([self.targetMenuView.customController isKindOfClass:OARouteDetailsGraphViewController.class])
```

Comment:
The problem was not in creating a variable, you should not cast anything here, the method you call should be in the customController protocol (better to make it optional and check for ```respondsToSelector```)